### DISCUSSION TITLE
numpy import error when using API

### DISCUSSION THREAD

**Discussion by kmulh-1** [juftin/camply#315] [General]
I developed a python app with the library on my mac using vscode and it works fine.  If I move the .py file to my raspberry pi, pip install camply and then run the app I get the following error.

ImportError: Unable to import required dependencies:
numpy: Error importing numpy: you should not try to import numpy from
        its source directory; please exit the numpy source tree, and relaunch
        your python interpreter from there.

Is this a problem with the way numpy is imported in the camply library or likely a setup problem on my side?

Thanks - Ken.

**Comment by juftin** [comment]
Hey @kmulh-1, this definitely isn't an issue with the code in camply but more likely an issue with the version of numpy camply uses: https://github.com/juftin/camply/blob/879caf66b6fc76199abff30fd7b1a84307db7f2a/pyproject.toml#L23-L24 (this pinning of numpy change is new as of an hour or so ago)

This probably means that numpy doesn't have an available binary for the RaspberryPi platform you're using. What Python version are you installing camply into?

**Reply by juftin** [reply]
🤷‍♂️ I should have this resolved in camply `0.32.1` - if you upgrade camply and rebuild your virtual environment that should resolve it for you https://github.com/juftin/camply/releases/tag/v0.32.1

**Reply by kmulh-1** [reply]
Thanks.  I'm running python 3.9 on the pi.  I'll upgrade camply later tonight and thanks for the quick fix!  I wrote my own library but it only supports reserveohio and reservations.gov so I'm switching it all over to your api.  I wrote my own app to support the whitelisting and blacklisting of individual sites and notification via IFTTT so I can get in app messages as well as SMS.  Thanks for the awesome libarary.

**Reply by juftin** [reply]
My pleasure! Let me know if it all works after that

**Reply by kmulh-1** [reply]
I verified the pi is upgraded with camply 0.32.1 but am still getting that error about numpy.  Anything else I can try?

**Reply by juftin** [reply]
Installing scientific libraries like numpy / pandas can be tricky on raspberrypi, see some documentation here: https://numpy.org/devdocs/user/troubleshooting-importerror.html#raspberry-pi

Have you removed the numpy / pandas installations in your virtual environment and rebuilt it? What happens when you `pip install numpy`? What version is downloaded? Does it install from a wheel or build from source? Did any of this work on `camply<32`? If it continues to fail providing more stacktrace could be helpful - I'm still not sure if your error is from running code or simply installing the libraries.

If all of this fails with installing camply / pandas / numpy with pip you can also use conda

**Reply by kmulh-1** [reply]
I finally got it working.  I unstalled numpy and pandas with pip3 and then installed pandas with apt-get, which also installed all pre-reqs including numpy.  I first tried this with just numpy, but the rpi repos have an older version and pandas didn't like it.

**Reply by illbefrankwithyou** [reply]
I'm stoked to use camply and am trying to get it setup on my pi. Does anyone have some detailed instructions for setup and running scripts? I have it running on my laptop but am a n00b with linux and the pi. Thanks for any help.
